src/Stack2nix.hs

```diff
@@ -234,11 +237,32 @@ toNix Args{..} remoteUri baseDir StackConfig{..} =
             Success expr ->
               case takeFileName fname of
                 "hspec.nix" ->
-                  writeFile fname $ show $ prettyNix $ (addParam "stringbuilder" . stripNonEssentialDeps) expr
-                _ ->
-                  writeFile fname $ show $ prettyNix $ stripNonEssentialDeps expr
+                  writeFile fname $ show $ prettyNix $ (addParam "stringbuilder" . stripNonEssentialDeps False) expr
+                name ->
+                  case argTestPkg of
+                    Just pkg -> do
+                      let shouldTest = name == pkg <.> "nix"
```

Comment:
The latest push does this. I verified the resulting nix expression still works with cardano-sl (rev 97850b14) when `--test` is used.